maap/AWS.py

```diff
@@ -15,51 +15,44 @@ def __init__(self, requester_pays_endpoint, s3_signed_url_endpoint, earthdata_s3
         self._s3_signed_url_endpoint = s3_signed_url_endpoint
         self._logger = logging.getLogger(__name__)
 
+    def requests_response(_url: str, _headers: object):
+        try:
+            response = requests.get(
+                url=_url,
+                headers=_headers
+            )
+
+            if response:
+                return json.loads(response.text)
+            response.raise_for_status()
+        except requests.exceptions.ConnectionError as e:
+            return ("Connection Error:", e)
+        except requests.exceptions.HTTPError as e:
+            return ("HTTP Error:", e)
+        except requests.exceptions.Timeout as e:
+            return ("Request Timeout:", e)
+        except requests.exceptions.TooManyRedirects as e:
+            return ("Redirects Error:", e)
+        except requests.exceptions.RequestException as e:
+            return ("Request Exception:", e)
+
     def requester_pays_credentials(self, expiration=60 * 60 * 12):
         headers = self._api_header
         headers['Accept'] = 'application/json'
 
-        response = requests.get(
-            url=self._requester_pays_endpoint + '?exp=' + str(expiration),
-            headers=self._api_header
-        )
-
-        if response:
-            return json.loads(response.text)
-        else:
-            return None
+        return self.requests_response(self._requester_pays_endpoint + '?exp=' + str(expiration), self._api_header)
```

Comment:
No need for a new method. I'd suggest dropping the new `requests_response` method and simply do the following in each of the existing methods:

```
response = requests.get(
    # existing args
)
response.raise_for_status()

return json.loads(response.text)
```